libraries/AC_PID/AC_P_1D.cpp

```diff
@@ -20,46 +20,60 @@ AC_P_1D::AC_P_1D(float initial_p, float dt) :
     AP_Param::setup_object_defaults(this, var_info);
 
     _kp = initial_p;
-    _lim_D_Out = 10.0f;     // maximum first differential of output
+    _error_min = 0.0f;
```

Comment:
No need to set these values to zero.  They will be zero by default I think.  Right @peterbarker?

Reply:
@peterbarker Can you confirm?

Reply:
Yes.  Unless you're ever constructing one of these on the stack you do not need to initialise them.

Here's the summary: https://ardupilot.org/dev/docs/style-guide.html#implicit-zeroing-of-memory

Reply:
Fixed in a few placed in the PID libary.